### PR TITLE
feat(guided-tour): add guided-tour package

### PR DESCRIPTION
**Pull request by christianmemije**
## Description

- Moved in guided tour package
- Added some demos

### TODOS:
- Documentation ✅ 
- More complex demos  ✅ 
- Placed under dialogs section, should it be it's own category?  ✅

Addresses #1696 
Addresses #1719 

#### Test Steps

- [ ] `npm ci`
- [ ] `npm run serve`
- [ ]  Check out http://localhost:4200/#/components/guided-tour/overview
- [ ] Verify build is successful  

#### General Tests for Every PR

- [ ] `npm run serve:prod` still works.
- [ ] `npm run tslint` passes.
- [ ] `npm run stylelint` passes.
- [ ] `npm test` passes and code coverage is not lower.
- [ ] `npm run build:lib` still works.

##### Screenshots or link to StackBlitz/Plunker

Screen Shot 2020-05-05 at 12 22 24
Screen Shot 2020-05-05 at 12 22 28
Screen Shot 2020-05-05 at 12 22 32

### CODE REVIEW

**Comment by emoralesb05** [comment]
Lets also add the nightly build commit

File: src/platform/guided-tour/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
++ initializeOnQueryParams(queryParam: string = 'tour'): Observable<ParamMap>
+  + Listen to query params to launch a tour
+
+```ts
+// for reference
+export type TourStep = Shepherd.Step.StepOptions;
+export type TourStepButton = Shepherd.Step.StepOptionsButton;
+
+export interface ITourEventOn {
+  selector?: string;
```
Comment (thomaslennan) [inline src/platform/guided-tour/README.md:22]: These properties are unique to our wrapper? If so, can we put a comment on each to explain usage?

File: src/app/content/components/component-demos/guided-tour/demos/guided-tour-demo-advance-on/guided-tour-demo-advance-on.component.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+export class GuidedTourDemoAdvanceOnComponent implements OnInit {
+  constructor(private _guidedTourService: CovalentGuidedTourService) {}
+
+  ngOnInit(): void {
+    const advanceOnTour: IGuidedTour = {
+      useModalOverlay: false,
+      steps: [
+        {
+          title: 'Step 1',
+          attachTo: {
```
Comment (thomaslennan) [inline src/app/content/components/component-demos/guided-tour/demos/guided-tour-demo-advance-on/guided-tour-demo-advance-on.component.ts:18]: Maybe add text here pointing user to click on button. And anchor step to the advance-button id?

File: src/app/content/components/component-demos/guided-tour/demos/guided-tour-demo-delay/guided-tour-demo-delay.component.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+export class GuidedTourDemoDelayComponent implements OnInit {
+  constructor(private _guidedTourService: CovalentGuidedTourService) {}
+
+  ngOnInit(): void {
+    const delayTour: IGuidedTour = {
+      useModalOverlay: true,
+      steps: [
+        {
+          title: 'Step 1',
+          attachTo: {
```
Comment (thomaslennan) [inline src/app/content/components/component-demos/guided-tour/demos/guided-tour-demo-delay/guided-tour-demo-delay.component.ts:18]: Same text here as in step 2?

Comment (christianmemije) [reply src/app/content/components/component-demos/guided-tour/demos/guided-tour-demo-delay/guided-tour-demo-delay.component.ts:18]: Can you clarify?

File: src/platform/guided-tour/README.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+A wrapper around [Shepherd](https://shepherdjs.dev) with extra functionality. Make sure to check out the [official shepherd docs](https://shepherdjs.dev/docs).
+
+## API Summary
+
+#### Methods
+
++ registerTour(tourName: string, tour: IGuidedTour | string): Promise<void>
+  + Create a tour
++ startTour(tour: string): Shepherd.Tour
```
[the remark this answers is not in the document]
Comment (christianmemije) [reply src/platform/guided-tour/README.md]: Done.

File: src/app/content/components/component-demos/guided-tour/demos/guided-tour-demo-advance-on/guided-tour-demo-advance-on.component.ts (hunk 2 above)
Comment (christianmemije) [reply src/app/content/components/component-demos/guided-tour/demos/guided-tour-demo-advance-on/guided-tour-demo-advance-on.component.ts:19]: Done.

File: src/platform/guided-tour/README.md (hunk 1 above)
Comment (christianmemije) [reply src/platform/guided-tour/README.md]: Done

File: src/platform/guided-tour/README.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  useModalOverlay: true,
+  steps: [
+    {
+      title: 'Fuel',
+      text: 'Here are the fuel levels',
+      attachTo: {
+        element: '#basic-demo #fuel',
+        on: 'top',
+      },
+    },
```
Comment (kriswinbush) [inline src/platform/guided-tour/README.md:99]: We could add all the options to the tour steps here. attachToOptions, advanceOnOptions, etc

Comment (christianmemije) [reply src/platform/guided-tour/README.md:99]: I think there's way too many options to list here 😭 . I wanted to just document a simple usage. More advanced usage can be found in the demos.

**Pull request merged**